File: Atcoder_#350/ruby/a_past_abcs.rb
s = gets.chomp

# sの末尾3文字を取得
number_str = s[3..5]

# 結合した文字列を整数に変換
number = number_str.to_i

# 範囲チェック
if (1..349).include?(number) && number != 316
  puts "Yes"
else
  puts "No"
end

File: Atcoder_#350/javascript/a_past_abcs.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('', (s) => {
  // sの末尾3文字を取得
  const numberStr = s.slice(3, 6);

  // 結合した文字列を整数に変換
  const number = parseInt(numberStr, 10);

  // 範囲チェック
  if (number >= 1 && number <= 349 && number !== 316) {
    console.log("Yes");
  } else {
    console.log("No");
  }

  rl.close();
});
